cowprotocol/cowswap
# feat(permit): allowance warning

# Summary

Fixes #3146

Update allowance warning check for limit orders

Now the UI will take into account permits still not executed when calculating whether the allowance warning should be displayed:

Before:
![image](https://github.com/cowprotocol/cowswap/assets/43217/29b828f0-28ce-42b2-876f-8cc5b92179c1)

After:
![image](https://github.com/cowprotocol/cowswap/assets/43217/e792be61-38cc-4130-8b20-c80e19809a80)


## TODO:

- [x] Move to jotai/updater
- [x] Check periodically
- [ ] Test more use cases
- [ ] Add it to Activity modal (SWAP orders)

  # To Test

1. Make sure there's no allowance
2. Place a LIMIT order for a permittable token
* Allowance warning should NOT be displayed
